web3c/oasis.js

```diff
@@ -154,9 +143,76 @@ class Oasis {
     this.accounts = accounts;
   }
 
+  _setupSubscribe(web3) {
+    this.subscribe = (subscription, options, callback) => {
+      // Only try to decrypt logs.
+      if (subscription !== 'logs') {
+        return web3.eth.subscribe(subscription, options, callback);
+      }
+
+      let wrappedEmitter = new EventEmitter();
+
+      let decryptSubscriptionLog = async (msg, log) => {
```

Comment:
should this be factored out to a method? it doesn't depend on the specific subscribe call, so by declaring it once you're helping the interpreter know the same function byte code can be used across calls

Reply:
This method should only be called once, so I'm guessing there's no saving? The only reason the method is here is for localizing all the subscribe related changes for readability/cleanliness.  If it's not serving that purpose lets move it.

Reply:
i don't have a strong preference either way.

Reply:
Ok I'll leave as is, then.